pkg/cluster/cluster.go

```diff
@@ -289,17 +278,56 @@ func New(ctx context.Context, log *logrus.Entry, _env env.Interface, db database
 		rpPrivateLinkServices:    network.NewPrivateLinkServicesClient(_env.Environment(), _env.SubscriptionID(), msiAuthorizer),
 		armRPPrivateLinkServices: armRPPrivateLinkServices,
 
-		dns:     dns.NewManager(_env, fpCredRPTenant),
-		storage: storage,
-		subnet:  subnet.NewManager(_env.Environment(), r.SubscriptionID, fpAuthorizer),
-		graph:   graph.NewManager(_env, log, aead, storage),
-		rpBlob:  rpBlob,
-
+		dns:                                    dns.NewManager(_env, fpCredRPTenant),
+		storage:                                storage,
+		subnet:                                 subnet.NewManager(_env.Environment(), r.SubscriptionID, fpAuthorizer),
+		graph:                                  graph.NewManager(_env, log, aead, storage),
+		rpBlob:                                 rpBlob,
 		installViaHive:                         installViaHive,
 		adoptViaHive:                           adoptByHive,
 		hiveClusterManager:                     hiveClusterManager,
 		now:                                    func() time.Time { return time.Now() },
 		openShiftClusterDocumentVersioner:      new(openShiftClusterDocumentVersionerService),
 		platformWorkloadIdentityRolesByVersion: platformWorkloadIdentityRolesByVersion,
-	}, nil
+	}
+
+	if doc.OpenShiftCluster.UsesWorkloadIdentity() {
+		err = m.platformWorkloadIdentityRolesByVersion.PopulatePlatformWorkloadIdentityRolesByVersion(ctx, doc.OpenShiftCluster, dbPlatformWorkloadIdentityRoleSets)
+		if err != nil {
+			return nil, err
+		}
+
+		msiResourceId, err := m.doc.OpenShiftCluster.ClusterMsiResourceId()
+		if err != nil {
+			return nil, err
+		}
+
+		msiDataplaneClientOptions, err := _env.MsiDataplaneClientOptions(msiResourceId)
+		if err != nil {
+			return nil, err
+		}
+
+		cloud, err := _env.Environment().CloudNameForMsiDataplane()
+		if err != nil {
+			return nil, err
+		}
+
+		authenticatorPolicy := dataplane.NewAuthenticatorPolicy(fpCredRPTenant, _env.MsiRpEndpoint())
+		msiDataplane, err := dataplane.NewClient(cloud, authenticatorPolicy, msiDataplaneClientOptions)
+		if err != nil {
+			return nil, err
+		}
+
+		clusterMsiKeyVaultName := _env.ClusterMsiKeyVaultName()
+		clusterMsiKeyVaultURL := fmt.Sprintf("https://%s.vault.azure.net", clusterMsiKeyVaultName)
```

Comment:
Can we use `_env.Environment().KeyVaultDNSSuffix` here instead of `vault.azure.net`??

Reply:
Done, and good point. I didn't realize there were different suffixes for different envs, so we might've hit issues in FairFax later on if you hadn't caught this.